Automattic/pocket-casts-ios
# Referrals: Entry point hint

| 📘 Part of: #2083  |  
|:---:|

Fixes # 

This PR implements the Referrals hint when accessing the Profile screen for the first time and referrals are availble.

For now we assuming that the hint is only show to a user until he taps on the gift icon for the first time. After that we save the status on the UserSettings and that user will not see it again on the same device.

| 1 | 2 | 3 | 4 | 5 |
| - | - | - | - | - |
| ![Simulator Screenshot - iPhone 15 Pro - 2024-09-02 at 16 13 25](https://github.com/user-attachments/assets/587f5288-669b-45a6-bedb-62a66c1b6372) | ![Simulator Screenshot - iPhone 15 Pro - 2024-09-02 at 16 13 16](https://github.com/user-attachments/assets/ed78bdc3-b9d1-45ac-9851-586341872d4e) | ![Simulator Screenshot - iPhone 15 Pro - 2024-09-02 at 16 13 05](https://github.com/user-attachments/assets/396f8278-2f20-460a-b764-1a06cc58123b) | ![Simulator Screenshot - iPhone 15 Pro - 2024-09-02 at 16 12 51](https://github.com/user-attachments/assets/bd2773a3-a2dd-448a-9c86-375258ab72fe) | ![Simulator Screenshot - iPhone 15 Pro - 2024-09-02 at 16 12 09](https://github.com/user-attachments/assets/dd2792c9-fd39-4c13-a34a-24c1b9f0713d) |

## To test

1. Start the app
2. Ensure that you have the Referrals FF enabled
3. Open the Profile view

### User with Plus

1. You should see the hint when you open profile
2. The hint should be dismissed if you press on any option on the screen or switch tab
3. If you come back the hint should still be visible
4. If you press on the gift icon the hint should be dismissed and you shouldn't see it again

### User without Plus

1. You shouldn't see the hint

## Checklist
